app/script/auth/page/Login.tsx

```diff
@@ -236,39 +236,44 @@ class Login extends React.Component<CombinedProps, State> {
           : this.props.doLogin(login);
       })
       .then(this.navigateChooseHandleOrWebapp)
-      .catch(error => {
-        switch (error.label) {
-          case BackendError.LABEL.NEW_CLIENT: {
-            this.props.resetAuthError();
-            /**
-             * Show history screen if:
-             *   1. database contains at least one event
-             *   2. there is at least one previously registered client
-             *   3. new local client is temporary
-             */
-            return this.props.doGetAllClients().then(clients => {
-              const shouldShowHistoryInfo = this.props.hasHistory || clients.length > 1 || !this.state.persist;
-              return shouldShowHistoryInfo
-                ? this.props.history.push(ROUTE.HISTORY_INFO)
-                : this.navigateChooseHandleOrWebapp();
-            });
-          }
-          case BackendError.LABEL.TOO_MANY_CLIENTS: {
-            this.props.resetAuthError();
-            return this.props.history.push(ROUTE.CLIENTS);
-          }
-          case BackendError.LABEL.INVALID_CREDENTIALS:
-          case LabeledError.GENERAL_ERRORS.LOW_DISK_SPACE: {
-            return;
-          }
-          default: {
-            const isValidationError = Object.values(ValidationError.ERROR).some(errorType =>
-              error.label.endsWith(errorType)
-            );
-            if (!isValidationError) {
-              throw error;
+      .catch((error: Error | BackendError) => {
```

Comment:
If you accept Error and BackendError as parameter of that function, you should handle both case

Reply:
I handle both cases. If it's a `BackendError` (detected by the `label` property), then it goes into the `switch` block. If it is a usual `Error`, then this error is thrown.